protocol/shannon/protocol.go

```diff
@@ -34,110 +32,93 @@ var (
 // This allows the protocol to report its sanctioned endpoints data to the devtools.DisqualifiedEndpointReporter.
 var _ devtools.ProtocolDisqualifiedEndpointsReporter = &Protocol{}
 
-// FullNode defines the set of capabilities the Shannon protocol integration needs
-// from a fullnode for sending relays.
+// GatewayClient provides all the methods needed by the Shannon protocol
+// package in PATH to interface with the Shannon Protocol.
 //
-// A properly initialized fullNode struct can:
-// 1. Return the onchain apps matching a service ID.
-// 2. Fetch a session for a (service,app) combination.
-// 3. Validate a relay response.
-// 4. Etc...
-type FullNode interface {
-	// GetApp returns the onchain application matching the application address
-	GetApp(ctx context.Context, appAddr string) (*apptypes.Application, error)
-
-	// GetSession returns the latest session matching the supplied service+app combination.
-	// Sessions are solely used for sending relays, and therefore only the latest session for any service+app combination is needed.
-	// Note: Shannon returns the latest session for a service+app combination if no blockHeight is provided.
-	GetSession(ctx context.Context, serviceID sdk.ServiceID, appAddr string) (sessiontypes.Session, error)
-
-	// GetAccountPubKey returns the account public key for the given address.
-	// The cache has no TTL, so the public key is cached indefinitely.
-	GetAccountPubKey(ctx context.Context, address string) (cryptotypes.PubKey, error)
-
-	// ValidateRelayResponse validates the raw bytes returned from an endpoint (in response to a relay request) and returns the parsed response.
-	ValidateRelayResponse(ctx context.Context, supplierAddr sdk.SupplierAddress, responseBz []byte) (*servicetypes.RelayResponse, error)
-
-	// IsHealthy returns true if the FullNode instance is healthy.
-	// A LazyFullNode will always return true.
-	// A CachingFullNode will return true if it has data in app and session caches.
-	IsHealthy() bool
+// It is implemented by the following concrete structs in the Shannon SDK package:
+//   - gatewayClient.CentralizedGatewayClient
```

Comment:
Gateway Mode concept, in its entirety, belongs on PATH side IMO.
The SDK can/should facilitate different operations related to any gateway mode, e.g. GetActiveSession for a particular service and App.

Reply:
I've updated this PR to move gateway mode clients into PATH. See commit https://github.com/buildwithgrove/path/pull/297/commits/e8838f9ad7ba6a58d5ad0233f417becd78afcb20